.github/workflows/compile-queries.yml

```
@@ -38,7 +38,9 @@ jobs:
         uses: actions/cache@v3
         with:
           path: '*/ql/src/.cache'
-          key: codeql-compile-main-${{ github.sha }} # just fill on main
+          key: codeql-compile-${{ github.ref }}-${{ github.sha }} # just fill on main
+          restore-keys: | # restore from another random commit, to speed up compilation.
+            codeql-compile-${{ github.ref }}- 
```

Would it make sense to add the `main` branch as a fall back?
```suggestion
            codeql-compile-${{ github.ref }}- 
            codeql-compile-refs/heads/main- 
```